### PR TITLE
Fine-tuning of PC-based SAM and NAO

### LINKED ISSUE #80: Fine-tuning of pc-based indices
- [x] #81
- [x] Adjust input for pc-based SAM: Either switch to Z500 or only limit input field to lat=-40..-65N, since slp for model data has valid values over land, opposed to real-world data. See #78.
- [ ] Extend definition for pc-based NAO: Make sure, index has correct sign. Therefore check related EOF pattern as specified location.

### LINKED ISSUE #81: Need covariance to be area-weighted: Multiply input field with sqrt(cos(lat))?

### PR DESCRIPTION
**Pull request by MarcoLandtHayen**
Just neglect minor differences and assume we have equidistant steps in latitude and longitude: Then the area-weighting can simply be done by multiplying input fields with sqrt(cos(latitude)). This is done in a separate function ```eof_weights``` and serves as alternative for the approach of @willirath. If accepted, closes #81 and #82.

In addition to that, we look at the leading EOF's positive pole and - if necessary - invert PC-based index values. This makes sure we have regular indices and PC-based indices to be positively correlated. Closes #80.

### CODE REVIEW

File: climate_index_collection/indices.py [hunk 1]
```diff
@@ -79,7 +81,9 @@ def southern_annular_mode_pc(data_set, geopoth_name="geopotential-height"):
 
     mask = data_set.coords["lat"] <= -20
 
-    geopoth = data_set[geopoth_name].where(mask)
+    geopoth = data_set[geopoth_name]
+    geopoth = geopoth * np.sqrt(np.cos(np.deg2rad(geopoth.lat)))
```
Comment (willirath) [inline climate_index_collection/indices.py:85]: I like this much simpler approach.

Still, I'd argue for indirecting the weight calculation to a function which can be reused in an easier way:

```python
def eof_weights(dobj):
    return np.sqrt(np.cos(np.deg2rad(dobj.coords["lat"])))
```
and then use, e.g., `geopoth = geopoth * eof_weights(geopoth.)`

Comment (MarcoLandtHayen) [reply climate_index_collection/indices.py]: Weight calculation is now done in a separate function, as you proposed.

File: climate_index_collection/reductions.py [hunk 2]
```diff
@@ -334,3 +334,22 @@ def area_mean_weighted(
     )
 
     return averaged
+
+
+def eof_weights(dobj):
+    """To have the Empirical Orthogonal Functions (EOFs) truly orthogonal, we need to take the area of the grid cells
```
Comment (willirath) [inline climate_index_collection/reductions.py:340]: I think this "truly orthogonal" argument is not accurate: The un-weighted EOFs are also forming a complete orthogonal basis. But the covariance matrix whose eigenvectors they are is not correct, if we don't weight the grid boxes according to the sample size they are averaged over.

**Review by willirath — COMMENTED** [review]
Looks good! I've left a comment clarifying the justification for the weighting.

File: climate_index_collection/reductions.py [hunk 3]
```diff
@@ -334,3 +334,22 @@ def area_mean_weighted(
...
+    into account. For equidistant latitude/longitude grids the area weights are proportional to cos(latitude).
```
Comment (willirath) [inline climate_index_collection/reductions.py:341]:
```suggestion
    """EOFs can be thought of as the eigen-vectors of the spatial covariance matrix. Grid cells can be thought of
    representing the mean over their spatial extent. They hence are low-pass filtering the raw signals and dampen the 
    variance by a factor proportional to the square root of the cell size.
    For equidistant latitude/longitude grids the area weights are proportional to cos(latitude).
```

**Pull request merged**